### DISCUSSION TITLE
"Band" effect with SelectiveBloomEffect.

### DISCUSSION THREAD

**Discussion by alexpineda** [pmndrs/postprocessing#407] [Q&A] [answered]
I am using DepthOfFieldEffect but in a different pass (before this). When I move the camera only a horizontal sliver gets bloom applied. 

![Recording 2022-09-03 at 11 50 17](https://user-images.githubusercontent.com/586716/188279012-6085ffd7-a6aa-41d2-8a2e-2325250e644f.gif)

**Comment by vanruesc** [comment, marked as answer]
Sounds like a depth issue. Does bloom work when you remove the pass that renders the `DepthOfFieldEffect`?

**Reply by alexpineda** [reply]
Yeah it's still doing it. `inverted` seems to do better though I have no understanding why.
![image](https://user-images.githubusercontent.com/586716/188284605-28a80dee-d4d4-4caa-b9a6-5c73038ba248.png)

**Reply by alexpineda** [reply]
Although with inverting the billboarded meshes don't seem to be affected. It's all a big mystery to me but I'm putting it down for now.

Edit: Never mind `inverted` is causing the screen to go black when I select units (sometimes). There is definitely some strange interaction happening. Additionally, I'm billboarding on gpu so makes sense it'll act strange for those (doh).

**Reply by vanruesc** [reply]
Selective bloom uses depth for masking so low depth precision can make this fail. What are your near/far camera settings? Also try setting `selectiveBloomEffect.depthMaskPass.epsilon` to `0.001` and `0.00001` and check if that does anything (default is `0.0001`).

If this only fails on specific devices, try enabling the `stencilBuffer` constructor parameter of the `EffectComposer` for guaranteed 24 bit depth.

**Reply by alexpineda** [reply]
near=0.1, far = 128 to 256

**Reply by alexpineda** [reply]
Well it's better now except a weird blacking out when I select a unit, some funny interaction there. I think I got caught with my bad practice of assigning camera, so now I'm creating new effects + effect passes on the changing of viewport. I also now set use override material since I'm using skinned meshes/instancing.
